### PR TITLE
feat(vpn): add resource VPN user

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
add resource VPN user
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add resource VPN user
```

## PR Checklist

* [x] Tests added/passed.

```
 make testacc TEST="./huaweicloud/services/acceptance/vpn" TESTARGS="-run TestAccUser_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccUser_basic -timeout 360m -parallel 4
=== RUN   TestAccUser_basic
=== PAUSE TestAccUser_basic
=== CONT  TestAccUser_basic
--- PASS: TestAccUser_basic (56.44s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       56.484s
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
   image

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
 image

### CODE REVIEW

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_user.go [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The user name.`,
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The user password.`,
+			},
```
Comment (houpeng80) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_user.go:60]: add Sensitive

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_user.go [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The user name.`,
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The user password.`,
```
Comment (saf3dfsa) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_user.go:60]: Sensitive:   true,

File: docs/resources/vpn_user.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used.
+  Changing this creates a new resource.
+
+* `vpn_server_id` - (Required, String, NonUpdatable) The VPN server ID.
```
Comment (ruwenqiang123) [inline docs/resources/vpn_user.md]:
```suggestion
* `vpn_server_id` - (Required, String, NonUpdatable) Specifies the VPN server ID.
```

Comment (ruwenqiang123) [reply docs/resources/vpn_user.md]: please check other requst parameters

**Pull request merged**